### PR TITLE
Camera package

### PR DESCRIPTION
**Pull request by Romanxz**

### CODE REVIEW

File: pages/camera.tsx [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+  const [photos, setPhotos] = useLocalStore("PhotoAlbum", []);
+  const [pickedImages, setPickedImages] = useLocalStore("Gallery", []);
+  const [images, setImages] = useLocalStore("Images", []);
+  const [deviceLinkId, setDeviceLinkId] = useLocalStore(
+    'deviceLinkId',
+    undefined
+  );
+
+  useEffect(() => {
+    const uploadPhotos = async (photos) => {
```
Comment (FreePhoenix888) [inline pages/camera.tsx:24]: Move this function to imports folder

File: imports/camera/initialize-package.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      data: [{
+        type_id: joinTypeLinkId,
+        to_id: await deep.id('deep', 'users', 'packages')
+      }, {
+        type_id: joinTypeLinkId,
+        to_id: await deep.id('deep', 'admin')
+      }]
+    },
+  })
+  const { data: [{ id: pacakgeTypeLinkId }] } = await deep.insert(PACKAGE_TYPES.map((TYPE) => ({
```
Comment (FreePhoenix888) [inline imports/camera/initialize-package.ts:32]: This is not pacakgeTypeLinkId, do not mislead other developers

**Review by FreePhoenix888 — CHANGES_REQUESTED** [review]

File: imports/camera/initialize-package.ts (hunk 2 above)
Comment (Romanxz) [reply imports/camera/initialize-package.ts]: now its packageTypes instead of packageType. I hope it makes sence but if not please suggest some name

**Review by FreePhoenix888 — APPROVED** [review]

File: pages/_app.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { defineCustomElements } from '@ionic/pwa-elements/loader';
+import { useEffect } from "react";
```
Comment (FreePhoenix888) [inline pages/_app.tsx:2]: Use defineCustomElements on your page in useEffect without dependencies:
```ts
  useEffect(() => {
    defineCustomElements(window);
  }, [])
  ```

**Review by FreePhoenix888 — CHANGES_REQUESTED** [review]

File: imports/camera/initialize-package.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    out: {
+      data: [{
+        type_id: joinTypeLinkId,
+        to_id: await deep.id('deep', 'users', 'packages')
+      }, {
+        type_id: joinTypeLinkId,
+        to_id: await deep.id('deep', 'admin')
+      }]
+    },
+  })
```
Comment (FreePhoenix888) [inline imports/camera/initialize-package.ts:31]: Add tree https://discord.com/channels/739430470345031692/997089753356845117/1083364263067914272

File: imports/camera/upload-photos.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+                },
+                {
+                    type_id: containTypeLinkId,
+                    to: {
+                        data: {
+                            type_id: webPathTypeLinkId,
+                            string: { data: { value: photo.webPath ? photo.webPath : "none" } },
+                        }
+                    }
+                },
```
Comment (FreePhoenix888) [inline imports/camera/upload-photos.ts:40]:
```suggestion
                photo.webPath && {
                    type_id: containTypeLinkId,
                    to: {
                        data: {
                            type_id: webPathTypeLinkId,
                            string: { data: { value: photo.webPath } },
                        }
                    }
                },
```

Do not create links if value is null/undefined.

_This comment applies to all the places working like these lines of code_

**Review by FreePhoenix888 — CHANGES_REQUESTED** [review]

File: imports/camera/take-photo.ts [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import { Camera, CameraResultType } from '@capacitor/camera';
+
+export default async function takePhoto() {
+  const image = await Camera.getPhoto({
+    quality: 90,
```
Comment (FreePhoenix888) [inline imports/camera/take-photo.ts:5]: Data used by capacitor must not be hardcoded and must be taken from deep

**Review by FreePhoenix888 — CHANGES_REQUESTED** [review]

**Pull request merged**